### PR TITLE
use @soroban-react npm library

### PR DESCRIPTION
**Pull request by esteblock**
Following https://github.com/stellar/soroban-example-dapp/issues/65 and https://github.com/stellar/soroban-example-dapp/issues/63

I created the [@soroban-react](https://github.com/esteblock/soroban-react) library based on this project's code, some commits https://github.com/stellar/soroban-example-dapp/pull/64 and insipred in the [web3-react](https://github.com/Uniswap/web3-react) library
See:
- https://www.npmjs.com/package/@soroban-react/core
- https://www.npmjs.com/package/@soroban-react/types
- https://www.npmjs.com/package/@soroban-react/freighter

```
yarn add @soroban-react/core
yarn add @soroban-react/types
yarn add @soroban-react/freighter
```

I implemented it in this PR

Even if for now only the Freighter wallet is configured, the  [@soroban-react](https://github.com/esteblock/soroban-react) library  allow to create more connectors very easily!

Both the library and the implementation need more work to do. But at least now the code is more clean and front end dev can focus on other things. 

Please open issues in https://github.com/esteblock/soroban-react/ to contribute to the library

### CODE REVIEW

**Comment by esteblock** [comment]
Don't know why is failing the test, but I think is because of https://github.com/stellar/soroban-example-dapp/pull/61

**Comment by esteblock** [comment]
Great! Check passed! Can you check it @paulbellamy ?
Thanks!

File: wallet/hooks/useNetwork.tsx [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
-import { AppContext } from "../AppContext";
-import { chain } from "../provideWalletChains";
+import { useSorobanReact } from "@soroban-react/core";import { chain } from "../provideWalletChains";
```
Comment (paulbellamy) [inline wallet/hooks/useNetwork.tsx:2]:
```suggestion
import { useSorobanReact } from "@soroban-react/core";
import { chain } from "../provideWalletChains";
```

**Review by paulbellamy — APPROVED** [review]
One style nitpick, but looks good. Thanks.

File: package.json [hunk 2]
```diff
@@ -12,6 +12,9 @@
   },
   "dependencies": {
     "@radix-ui/react-dialog": "1.0.2",
+    "@soroban-react/core": "^1.0.2",
+    "@soroban-react/freighter": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (esteblock) [reply package.json:16]: Great!

**Pull request merged**